### PR TITLE
BraidProvider: Add 'linkComponent' prop to customise links

### PR DESCRIPTION
**Pull request by markdalgleish**
See changeset for details.

### CODE REVIEW

File: lib/components/BraidProvider/BraidProvider.tsx [hunk 1]
```diff
@@ -1,5 +1,11 @@
-import React, { createContext, useContext, ReactNode } from 'react';
-import { TreatProvider } from 'sku/react-treat';
+import React, {
+  createContext,
+  useContext,
+  ReactNode,
+  ComponentType,
+  AnchorHTMLAttributes,
+} from 'react';
+import { TreatProvider } from 'sku/treat';
```
Comment (michaeltaranto) [inline lib/components/BraidProvider/BraidProvider.tsx:8]: This looks like an accidental change, should be importing from `sku/react-treat`.

Comment (markdalgleish) [reply lib/components/BraidProvider/BraidProvider.tsx:8]: 😫

File: lib/components/BraidProvider/BraidProvider.tsx [hunk 2]
```diff
@@ -20,31 +26,47 @@ export const useBraidTheme = () => {
   return braidTheme;
 };
 
+export interface LinkComponentProps
+  extends AnchorHTMLAttributes<HTMLAnchorElement> {
+  href: string;
+}
+export type LinkComponent = ComponentType<LinkComponentProps>;
+const DefaultLinkComponent = (props: LinkComponentProps) => <a {...props} />;
```
Comment (michaeltaranto) [inline lib/components/BraidProvider/BraidProvider.tsx:34]: Should the default be a reset anchor tag, eg. `<Box component="a" />`, thoughts? I cant decide.

File: lib/components/Link/Link.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+export type LinkProps = LinkComponentProps;
+
+export const Link = ({ href, className, ...restProps }: LinkComponentProps) => {
+  const LinkComponent = useLinkComponent();
+
+  return (
+    <LinkComponent
+      href={href}
+      className={classnames(resetStyles.base, resetStyles.element.a, className)}
```
Comment (michaeltaranto) [inline lib/components/Link/Link.tsx:17]: ah gotcha, just saw this.

any reason not to use `useBoxStyles({ component: 'a', className })`?

Comment (markdalgleish) [reply lib/components/Link/Link.tsx:17]: Yeah, that probably makes more sense 👍

File: site/src/App/Code/Code.tsx [hunk 4]
```diff
@@ -24,8 +24,8 @@ const formatSnippet = memoize(
   snippet =>
     prettier
       .format(snippet, {
-        parser: 'babel',
-        plugins: [babylonParser],
+        parser: 'typescript',
+        plugins: [typescriptParser],
```
Comment (michaeltaranto) [inline site/src/App/Code/Code.tsx:28]: 👏

File: lib/components/BraidProvider/BraidProvider.tsx (hunk 1 above)
Comment (markdalgleish) [reply lib/components/BraidProvider/BraidProvider.tsx]: This looks like a failed merge from master on my part.

**Pull request merged**